ports/rp2/rp2_pio.c

```diff
@@ -462,21 +462,32 @@ STATIC mp_obj_t rp2_state_machine_init_helper(const rp2_state_machine_obj_t *sel
     }
 
     // Compute the clock divider.
-    float div;
+    uint16_t clkdiv_int;
+    uint8_t clkdiv_frac;
     if (args[ARG_freq].u_int < 0) {
-        div = 1;
+        // Default: run at CPU frequency.
+        clkdiv_int = 1;
+        clkdiv_frac = 0;
     } else if (args[ARG_freq].u_int == 0) {
-        div = 0;
+        // Special case of 0: set clkdiv to 0.
```

Comment:
Do the MicroPython PIO docs (if they exist yet) mention these special-cases of `freq < 0` and `freq == 0` ?

Reply:
`freq < 0` is just the default case (if `freq` is not specified in the constructor it defaults to `-1`) and not necessarily something the user needs to know about (but they could pass in `-1` if they wanted...).

`freq == 0` is not documented.